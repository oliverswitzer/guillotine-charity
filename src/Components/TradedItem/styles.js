export default () => ({
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  productImage: {
    margin: 10
  }
});
