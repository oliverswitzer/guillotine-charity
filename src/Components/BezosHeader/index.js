import React from 'react';
import PropTypes from 'prop-types';

// Materail-UI
import { withStyles } from '@material-ui/styles';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import styles from './styles';
// -------

export const BezosHeaderRaw = ({ classes }) => (
  <Paper className={classes.paper}>
    <Grid container spacing={3} direction="column" justify="center" alignItems="center">
      <Grid item xs={12}>
        <Avatar alt="Jeff Bezos" src="/images/bezos.jpg" className={classes.avatar} />
      </Grid>
      <Grid item xs={12}>
        <Typography component="h1" variant="h4" align="center" color="textPrimary">
          Spend Jeff Bezos&apos; Money
        </Typography>
      </Grid>
    </Grid>
  </Paper>
);

BezosHeaderRaw.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired
};

export default withStyles(styles)(BezosHeaderRaw);
