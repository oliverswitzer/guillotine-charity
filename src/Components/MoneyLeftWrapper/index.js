import React from 'react';
import PropTypes from 'prop-types';

// Material-UI
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/styles';
import styles from './styles';
// -------

import MoneyStringConverter from '../../Utils/MoneyStringConverter';

export const MoneyLeftWrapperRaw = ({ classes, moneyLeft }) => (
  <div className={classes.moneyWrapper}>
    <Typography component="h1" variant="h4" align="center" noWrap color="textPrimary">
      {`$${MoneyStringConverter(moneyLeft)} Left`}
    </Typography>
  </div>
);

MoneyLeftWrapperRaw.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  moneyLeft: PropTypes.number.isRequired
};

export default withStyles(styles)(MoneyLeftWrapperRaw);
