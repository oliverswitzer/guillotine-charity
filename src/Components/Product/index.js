import React from 'react';
import PropTypes from 'prop-types';

// Material-UI
import { withStyles } from '@material-ui/styles';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Input from '@material-ui/core/Input';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import styles from './styles';
// -------

import MoneyStringConverter from '../../Utils/MoneyStringConverter';

export class ProductRaw extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      quantity: 0
    };
  }

  // Since setState is asyncronous, this method is always called in the setState's callback.
  // So that this.state.quantity will hold the correct updated value when we call onTradedItem.
  onProductUpdated = () => {
    const { item, onTradedItem } = this.props;
    const { quantity } = this.state;
    const itemTraded = {
      ...item,
      quantity
    };
    onTradedItem(itemTraded);
  };

  onChangeQuantity = event => {
    if (event.target.value > 0) {
      this.setState(
        {
          quantity: Number(event.target.value)
        },
        this.onProductUpdated
      );
    }
  };

  onButtonClick = value => () => {
    const { quantity } = this.state;
    this.setState(
      {
        quantity: quantity + value
      },
      this.onProductUpdated
    );
  };

  render() {
    const { classes, item } = this.props;
    const { quantity } = this.state;
    const enableSellButton = quantity === 0;
    return (
      <Card className={classes.card}>
        <CardMedia image={item.image} title={item.name} className={classes.cardMedia} />
        <CardContent className={classes.cardContent}>
          <Typography variant="h5" component="h2">
            {item.name}
          </Typography>
          <Typography variant="h5" component="h2">
            {`$${MoneyStringConverter(item.price)} USD`}
          </Typography>
        </CardContent>
        <CardActions>
          <Button
            size="medium"
            color="secondary"
            onClick={this.onButtonClick(-1)}
            disabled={enableSellButton}
          >
            Sell
          </Button>
          <Input
            className={classes.input}
            type="number"
            label="quantity"
            inputProps={{ shrink: 'True' }}
            margin="dense"
            placeholder="Quantity"
            value={quantity}
            onChange={this.onChangeQuantity}
          />
          <Button size="medium" color="primary" onClick={this.onButtonClick(1)}>
            Buy
          </Button>
        </CardActions>
      </Card>
    );
  }
}

ProductRaw.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  item: PropTypes.shape({
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired,
  onTradedItem: PropTypes.func.isRequired
};

export default withStyles(styles)(ProductRaw);
