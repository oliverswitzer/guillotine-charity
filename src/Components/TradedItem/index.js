import React from 'react';
import PropTypes from 'prop-types';

// Material-UI
import { withStyles } from '@material-ui/styles';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import styles from './styles';
// -------

export const TradedItemRaw = ({ item, classes }) => (
  <div className={classes.container}>
    <Avatar src={item.image} alt={item.name} className={classes.productImage} />
    <Typography variant="h5" component="h5" color="textPrimary">
      {`${item.name} x ${item.quantity}`}
    </Typography>
  </div>
);

TradedItemRaw.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  item: PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    quantity: PropTypes.number.isRequired
  }).isRequired
};

export default withStyles(styles)(TradedItemRaw);
