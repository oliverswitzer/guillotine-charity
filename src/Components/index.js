import BezosHeader from './BezosHeader';
import MoneyLeftWrapper from './MoneyLeftWrapper';
import Product from './Product';
import TradedItem from './TradedItem';

export default {
  BezosHeader,
  MoneyLeftWrapper,
  Product,
  TradedItem
};
