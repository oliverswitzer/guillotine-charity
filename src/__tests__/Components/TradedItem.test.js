import React from 'react';
import renderer from 'react-test-renderer';
import { TradedItemRaw as TradedItem } from '../../Components/TradedItem';

describe('TradedItem Component', () => {
  it('matches snapshot', () => {
    const props = {
      classes: {
        container: 'container-css',
        productImage: 'product-image-css'
      },
      item: {
        name: 'dummy item name',
        quantity: 1,
        image: '/path/to/dummy/traded/item'
      }
    };
    const tradedItemComponentTree = renderer.create(<TradedItem {...props} />).toJSON();
    expect(tradedItemComponentTree).toMatchSnapshot();
  });
});
