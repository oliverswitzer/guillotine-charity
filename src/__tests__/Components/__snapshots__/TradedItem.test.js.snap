// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradedItem Component matches snapshot 1`] = `
<div
  className="container-css"
>
  <div
    className="MuiAvatar-root product-image-css"
  >
    <img
      alt="dummy item name"
      className="MuiAvatar-img"
      src="/path/to/dummy/traded/item"
    />
  </div>
  <h5
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextPrimary"
  >
    dummy item name x 1
  </h5>
</div>
`;
